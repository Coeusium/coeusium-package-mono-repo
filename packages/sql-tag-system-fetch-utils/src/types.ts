export type { Tag } from 'sql-tag-system';
