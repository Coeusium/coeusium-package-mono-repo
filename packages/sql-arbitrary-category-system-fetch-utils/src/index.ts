export * from './sql-arbitrary-category-system-fetch-utils';
export type { Category, CategorySet } from './types';
