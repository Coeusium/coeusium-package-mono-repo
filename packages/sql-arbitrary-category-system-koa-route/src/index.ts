export * from './sql-arbitrary-category-system-koa-route';
export type { ConnectionInfo, Category, CategorySet } from './types';
