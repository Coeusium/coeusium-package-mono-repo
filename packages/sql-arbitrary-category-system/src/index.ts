export * from './sql-arbitrary-category-system';
export type { ConnectionInfo, Category, CategorySet } from './types';
