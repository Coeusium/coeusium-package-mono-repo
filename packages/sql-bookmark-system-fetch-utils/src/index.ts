export * from './sql-bookmark-system-fetch-utils';
export type { Bookmark } from './types';
