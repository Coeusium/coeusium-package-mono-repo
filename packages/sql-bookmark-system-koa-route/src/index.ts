export * from './sql-bookmark-system-koa-route';
export type { ConnectionInfo, Bookmark } from './types';
