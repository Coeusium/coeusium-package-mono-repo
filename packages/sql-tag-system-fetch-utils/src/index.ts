export * from './sql-tag-system-fetch-utils';
export type { Tag } from './types';
