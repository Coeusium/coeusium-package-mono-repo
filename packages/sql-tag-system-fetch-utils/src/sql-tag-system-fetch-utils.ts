import { Tag } from './types';

export async function addTag(tag) {
  const res = await fetch(`/api/tag?tag=${tag}`, { method: 'post' });
  const { id } = await res.json();
  return id as string;
}

export async function getTag(id) {
  const res = await fetch(`/api/tag?id=${id}`);
  return (await res.json()) as Tag;
}

export async function listTags(page = 0, limit = 20) {
  const res = await fetch(`/api/tag?page=${page}&limit=${limit}`);
  return (await res.json()) as Tag[];
}

export async function deleteTag(id) {
  const res = await fetch(`/api/tag?id=${id}`, { method: 'delete' });
  await res.json();
}
