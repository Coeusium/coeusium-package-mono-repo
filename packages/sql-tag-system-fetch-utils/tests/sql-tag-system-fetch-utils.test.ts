import * as sqlTagSystemFetchUtils from '../src';

describe('sql-tag-system-fetch-utils', () => {
  it('Has Correct API', () => {
    const keys = Object.keys(sqlTagSystemFetchUtils);
    expect(keys.length).toBe(4);
    expect(keys).toMatchSnapshot();
  });
});
