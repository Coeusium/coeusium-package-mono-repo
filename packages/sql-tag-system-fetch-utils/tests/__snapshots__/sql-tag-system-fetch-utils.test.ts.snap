// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sql-tag-system-fetch-utils Has Correct API 1`] = `
Array [
  "addTag",
  "getTag",
  "listTags",
  "deleteTag",
]
`;
